### PR TITLE
Investigating new ironic pod

### PR DESCRIPTION
**Pull request by namnx228**

### CODE REVIEW

File: ironic/ironic_deployment_investigation.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+There are 7 containers in an ironic pod:
+- ironic: This container has one ironic process. This process runs in the combined mode of ironic, as opposed to the separated mode where api and conductor parts run in two different processes.
+	+ Port 6388: Listen to api requests. These requests are handled by api part of ironic. In fact, the traffic coming to this port is mostly from the reverse proxy ironic-httpd.
+- ironic-inspector: not much change
+	+ Port 5049: Listen to inspector requests. The traffic coming to this port is mostly from the reverse proxy ironic-httpd.
+- ironic-httpd:
+	+ Port 6385: Any traffic coming to this port is forwarded to port 6388 of ironic container.
+	+ Port 5050: Any traffic coming to this port is forwarded to port 5049 of ironic-inspector container.
+	+ Port 6180: Allows downloading IPA image
+- mariadb: This database will so	on be replaced by SQLite by default.
```
Comment (kashifest) [inline ironic/ironic_deployment_investigation.md:13]: the line seems incomplete.

Comment (smoshiur1237) [inline ironic/ironic_deployment_investigation.md:13]: typo

Comment (namnx228) [reply ironic/ironic_deployment_investigation.md]: Fixed!. Thanks

File: ironic/ironic_deployment_investigation.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# New Ironic deployment investigation
+In Metal3, ironic is deployed in a container image, the repo of which can be found at: https://github.com/metal3-io/ironic-image. This document provides some details about the ports that an ironic pod in Metal3 cluster can open and the interaction between ports. The commit used for this investigation of ironic-image repo is **09ad5a22056a4334db76fd10b8def660936d1a03**.
+
+There are 7 containers in an ironic pod:
+- ironic: This container has one ironic process. This process runs in the combined mode of ironic, as opposed to the separated mode where api and conductor parts run in two different processes.
+	+ Port 6388: Listen to api requests. These requests are handled by api part of ironic. In fact, the traffic coming to this port is mostly from the reverse proxy ironic-httpd.
```
Comment (furkatgofurov7) [inline ironic/ironic_deployment_investigation.md:6]: Listen**s**

File: ironic/ironic_deployment_investigation.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- ironic-inspector: No change
+	+ Port 5049: Listen to inspector requests. The traffic coming to this port is mostly from the reverse proxy ironic-httpd.
```
Comment (furkatgofurov7) [inline ironic/ironic_deployment_investigation.md:8]: /s

File: ironic/ironic_deployment_investigation.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	+ Port 6385: Any traffic coming to this port is forwarded to port 6388 of ironic container.
+	+ Port 5050: Any traffic coming to this port is forwarded to port 5049 of ironic-inspector container.
+	+ Port 6180: Allows downloading IPA image
+- mariadb: This database will soon be replaced by SQLite by default.
+	+ Port 3306: Database port. Ironic connects to this port to read and write node data.
+- ironic-keepalived: No change
+- ironic-dnsmasq: No change
+	+ Port 67: bootstrap server. For PXE boot
+	+ Port 69: TFTP
+- Ironic-log-watch: No change
```
Comment (furkatgofurov7) [inline ironic/ironic_deployment_investigation.md:19]:
```suggestion
- ironic-log-watch: No change
```

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**